src/sub/payments.py

```diff
@@ -165,47 +168,108 @@ def delete_customer(uid: str) -> FlaskResponse:
     :param uid:
     :return: Success of failure message for the deletion
     """
+    logger.info("delete customer", uid=uid)
     subscription_user = g.subhub_account.get_user(uid)
-    if not subscription_user:
-        return dict(message="Customer does not exist."), 404
-
-    deleted_payment_customer = vendor.delete_stripe_customer(subscription_user.cust_id)
-    if deleted_payment_customer:
-        deleted_customer = delete_user(
-            user_id=subscription_user.user_id,
-            cust_id=subscription_user.cust_id,
-            origin_system=subscription_user.origin_system,
+    logger.info("delete customer", subscription_user=subscription_user)
+    if subscription_user is not None:
+        origin = subscription_user.origin_system
+        logger.info("delete origin", origin=origin)
+        if not subscription_user:
+            return dict(message="Customer does not exist."), 404
+        subscribed_customer = vendor.retrieve_stripe_customer(subscription_user.cust_id)
+        subscribed_customer = subscribed_customer.to_dict()
+        subscription_info: List = []
+        logger.info(
+            "subscribed customer",
+            subscribed_customer=subscribed_customer,
+            data_type=type(subscribed_customer),
         )
-        user = g.subhub_account.get_user(uid)
-        if deleted_customer and user is None:
-            return dict(message="Customer deleted successfully"), 200
-
+        for subs in subscribed_customer["subscriptions"]["data"]:
+            sub = dict(
+                plan_amount=subs.plan.amount,
+                nickname=subs.plan.nickname,
```

Comment:
This needs to use the `format_plan_nickname` method

Reply:
https://github.com/mozilla/subhub/issues/311

Reply:
@jackiemunroe updated, please review.

Reply:
The nickname formatter uses the Product.name. The Plan.nickname should not be one of the parameters.